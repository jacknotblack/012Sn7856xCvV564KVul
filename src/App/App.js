import React, { Component } from "react";
import PropTypes from "prop-types";
import { Route, withRouter } from "react-router";
import withUserAgent from "react-useragent";
import { connect } from "react-redux";
import "./App.css";
import appActions from "./actions";
import Home from "../components/home";
import LoadableRanks from "../modules/ranks/index.lazy";
import LoadableBilling from "../modules/billing/index.lazy";

class App extends Component {
  componentWillMount() {
    this.props.setWebview(this.props.isWebview());
  }

  render() {
    return (
      <div className="App">
        <Route exact path="/" component={Home} />
        <Route
          path="/ranks"
          render={match => <LoadableRanks {...{ match }} />}
        />
        <Route
          path="/billing"
          render={match => <LoadableBilling {...{ match }} />}
        />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  setWebview: isWebview => {
    dispatch(appActions.setWebviewMode(isWebview));
  },
  isWebview: userAgent => /SENSESTAR/.test(userAgent)
});

App.propTypes = {
  setWebview: PropTypes.func.isRequired,
  isWebview: PropTypes.func.isRequired
};

export const AppConnected = connect(null, mapDispatchToProps)(
  withUserAgent(App)
);
export default withRouter(AppConnected);
export { App as Pure };
