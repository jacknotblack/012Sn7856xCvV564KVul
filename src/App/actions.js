const appActions = {
  setWebviewMode: isWebview => ({
    type: "SET_WEBVIEW_MODE",
    payload: isWebview
  })
};

export default appActions;
