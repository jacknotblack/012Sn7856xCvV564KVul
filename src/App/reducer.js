const initState = {
  isWebview: false
};

const rankReducer = (state = initState, action) => {
  switch (action.type) {
    case "SET_WEBVIEW_MODE":
      return Object.assign({}, state, {
        isWebview: action.payload
      });
    default:
      return state;
  }
}; // NOSONAR

export default rankReducer;
