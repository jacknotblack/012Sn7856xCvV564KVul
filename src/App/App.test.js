import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Route } from "react-router";
import { createMockStore } from "redux-test-utils";
import { AppConnected, Pure } from "./App";
import Home from "../components/home";
import LoadableRanks from "../modules/ranks/index.lazy";

describe("Desktop", () => {
  describe("render", () => {
    let wrapper;
    const props = {
      isWebview: jest.fn(),
      setWebview: jest.fn(),
      ua: {}
    };

    beforeEach(() => {
      wrapper = shallow(<Pure {...props} />);
    });

    it("renders without crashing", () => {
      expect(wrapper.length).toEqual(1);
    });

    it("should match snapshot", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders correct routes", () => {
      const match = "/ranks";

      /* eslint-disable no-param-reassign */
      const pathMaps = wrapper.find(Route).reduce((pathMap, route) => {
        const routeProps = route.props();
        pathMap[routeProps.path] =
          routeProps.component || routeProps.render(match);
        return pathMap;
      }, {});
      /* eslint-enable no-param-reassign */

      expect(pathMaps["/"]).toBe(Home);
      expect(pathMaps["/ranks"]).toEqual(<LoadableRanks {...{ match }} />);
    });

    it("execute setWebview and isWebview when _willmount executed", () => {
      expect(props.setWebview).toHaveBeenCalled();
      expect(props.isWebview).toHaveBeenCalled();
    });
  });

  describe("container", () => {
    const shallowWithStore = (component, store) => {
      const context = {
        store
      };
      return shallow(component, { context });
    };
    const testState = {
      isWebview: true
    };
    const store = createMockStore(testState);
    const container = shallowWithStore(<AppConnected />, store);

    it("should render successfully", () => {
      expect(container).toBeTruthy();
    });

    it("should dispatch when calling action", () => {
      const props = {
        userAgent: {
          nonWebview: {
            mobile: null,
            os: null,
            phone: null,
            tablet: null,
            md: {
              maxPhoneWidth: 600,
              ua: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) 
            AppleWebKit/537.36 (KHTML, like Gecko) 
            Chrome/62.0.3202.94 Safari/537.36`
            }
          },
          webview: {
            mobile: null,
            os: null,
            phone: null,
            tablet: null,
            md: {
              maxPhoneWidth: 600,
              ua: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) 
            AppleWebKit/537.36 (KHTML, like Gecko) 
            Chrome/62.0.3202.94 Safari/537.36/SENSESTAR/`
            }
          }
        }
      };
      expect(
        container.props().isWebview(props.userAgent.webview.md.ua)
      ).toEqual(true);
      expect(
        container.props().isWebview(props.userAgent.nonWebview.md.ua)
      ).toEqual(false);
      container.props().setWebview(false);
      expect(
        store.isActionDispatched({
          type: "SET_WEBVIEW_MODE",
          payload: false
        })
      ).toEqual(true);
    });
  });
});
