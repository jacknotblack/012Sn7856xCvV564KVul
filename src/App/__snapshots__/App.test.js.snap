// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop render should match snapshot 1`] = `
<div
  className="App"
>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    path="/ranks"
    render={[Function]}
  />
  <Route
    path="/billing"
    render={[Function]}
  />
</div>
`;
